File: yunshin/4주차/BruteForce/Programmers/42839_소수찾기.js
var splited,
  numList = [],
  visited = []
var Composit = new Array(9999999 + 1).fill(false)

function dfs(curNum) {
  if (curNum != '') {
    numList.push(Number(curNum))
  }

  for (let i = 0; i < splited.length; i++) {
    if (!visited[i]) {
      visited[i] = true
      dfs(curNum + splited[i])
      visited[i] = false
    }
  }
}

// 에라토스테네스의 체
function isComposit() {
  Composit[0] = true
  Composit[1] = true

  for (let i = 2; i < Math.pow(Composit.length, 1 / 2); i++) {
    for (let j = 2; i * j < Composit.length; j++) {
      Composit[i * j] = true
    }
  }
}

function solution(numbers) {
  splited = numbers.split('')
  for (i = 0; i < splited.length; i++) {
    visited.push(false)
  }
  dfs('')
  numList = [...new Set(numList)]
  console.log(numList)
  isComposit()
  return numList.filter((x) => !Composit[x]).length
}

File: yunshin/4주차/BruteForce/Programmers/42839_소수찾기.py
# DFS, 에라토스테네스의 체
def isPrime(x):
    if x == 1:
        return False
    for i in range(2, int(x**(1/2))+1):
        if x%i == 0:
            return False
    return True

def solution(numbers):
    global numList, visited, numsSet
    numList = list(numbers)
    visited = [False] * len(numbers)
    numSet = set()
    def dfs(res):
        if len(res)<= len(numList):
            if res and res[0] =="0":
                return
            numSet.add(res)
        for i in range(len(numList)):
            if not visited[i]:
                visited[i] = True
                dfs(res + (numList[i]))
                visited[i] = False
    dfs("")
    answer = 0 
    numList = list(numSet)
    numList.remove("")
    for x in list(numList):
        if(isPrime(int(x))):
            answer+=1
    return answer